multimedia/VLC-X11/Portfile

```diff
@@ -0,0 +1,357 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:filetype=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem              1.0
+PortGroup               compiler_blacklist_versions 1.0
+PortGroup               legacysupport 1.1
+PortGroup               openssl 1.0
+
+# At the moment we track releases of VLC.
+# As long as everything works on 10.5+,
+# no version pegging is needed.
+name                    VLC-X11
+version                 3.0.20
+revision                0
+categories              multimedia
+license                 GPL-2 LGPL-2.1
+maintainers             {@barracuda156 gmail.com:vital.had} openmaintainer
+
+description             VLC media player with X11 backend
+long_description        {*}${description}, aimed primarily at supporting legacy \
+                        macOS, including PowerPC systems. It may also be used \
+                        on modern macOS versions, as long as GUI is unneeded.
+homepage                https://www.videolan.org
+master_sites            https://download.videolan.org/pub/videolan/vlc/${version}/
+distname                vlc-${version}
+checksums               rmd160  0a6263bbe189a2c08d2c2f5c94ee4bee908cb716 \
+                        sha256  adc7285b4d2721cddf40eb5270cada2aaa10a334cb546fd55a06353447ba29b5 \
+                        size    25596392
+use_xz                  yes
+
+use_autoreconf          yes
+
+# The purpose of this port is to be as compatible as possible and work on every macOS
+# version, including PowerPC systems. Priority is given to legacy systems,
+# since on the current ones VLC3 can be built with GUI and all modules.
+# We do not want a dozen versions here with complex conditional fallback code.
+# Some choices below may not be the first best for newer macOS versions.
+# If this does not fit your aims, please rather install a standard VLC3
+# either from Macports or the developer.
+
+# FIXME: what is labelled bonjour in VLC3 has no relation to bonjour in VLC2,
+# where it was an Avahi module. VLC3 has some ObjC sourcery instead,
+# which unsurprisingly fails to compile. libsecuretransport did not build
+# already in VLC 2.2.8. However, it may be possible to fix both.
+
+# Macros are modified to be actually useful: HAVE_OSX becomes a configurable option
+# instead of a coerced one; HAVE_DARWIN is used as a macOS baseline.
+# That is, everything macOS-specific which can be supported from 10.5+
+# should go under HAVE_DARWIN. Anything which requires modern Xcode
+# or newer SDK and does not have a fallback implemented should go inside HAVE_OSX,
+# and we conveniently disable all that breaking stuff via --disable-osx arg.
+# Notice, this is not the same as --disable-macosx* of the upstream;
+# despite a suggestive wording, the latter fails to actually disable OSX code.
```

Comment:
```suggestion
# despite a suggestive wording, the latter fails to actually disable macOS code.
```

Reply:
Commit title is wrong, but I will squash these anyway. (Just woke up and reviewing this from mobile.)